### DISCUSSION TITLE
Getting the compilation status from a vscode extension

### DISCUSSION THREAD

**Discussion by germainaubert** [redhat-developer/vscode-java#2094] [Q&A] [answered]
Hello !

I'm working on a vscode extension in which I have to get the compilation status of some java code written in vscode.

I have been looking over the vscode-java extension interface but couldn't find any function that might help me.

Is there a way to get such an information from the vscode-java extension?

Thank you in advance!

**Comment by germainaubert** [comment, marked as answer]
Update: I found the solution.

All I had to do is add the *vscode-java* extension in the extension dependencies (package.json).
Then I was able to call the compile command using ```vscode.commands.executeCommand```.

**Reply by Eskibear** [reply]
available APIs : https://github.com/redhat-developer/vscode-java/blob/master/src/extension.api.ts

**Reply by rgrunber** [reply]
In case anyone needs this in the future, the compile command is [`"java.workspace.compile"`](https://github.com/redhat-developer/vscode-java/blob/81482a22f966580a66bdce9745116379e0ebd194/src/commands.ts#L80) passed as the command argument to `vscode.commands.executeCommand`. It should return a [`CompileWorkspaceStatus`](https://github.com/redhat-developer/vscode-java/blob/81482a22f966580a66bdce9745116379e0ebd194/src/protocol.ts#L65)
